packages/@aws-cdk/aws-apigatewayv2/lib/websocket/stage.ts

```diff
@@ -87,14 +112,103 @@ export class WebSocketStage extends StageBase implements IWebSocketStage {
+     */
+    const rateLimits = !props.throttle ? undefined : {
+      throttlingBurstLimit: !props.throttle.burstLimit ? undefined : props.throttle.burstLimit,
+      throttlingRateLimit: !props.throttle.rateLimit ? undefined : props.throttle.rateLimit,
+    };
+
+    /**
+     * The data trace log settings for CfnStage use two parameters, dataTraceEnabled to
+     * turn on the logging and a logging level.  However, specifing a logging level of
+     * OFF is the same as disabling data tracing and setting it to INFO or ERROR implies
+     * that data tracing is on, so the value of the logging flag is being deduced from
+     * the value of the log level.
+     */
+    const dataTraceLoggingSettings =
+      (!props.dataTraceLoggingLevel || props.dataTraceLoggingLevel === 'OFF') ?
+        undefined :
+        {
+          dataTraceEnabled: true,
+          loggingLevel: props.dataTraceLoggingLevel,
+        };
+
+    // If the user has specfied a retention time and/or removal policy for the log group use it,
+    // otherwise make it a 30 day retention and a removal policy of DESTROY.
+    if (props.dataTraceLoggingLevel && props.dataTraceLoggingLevel !== 'OFF') {
+      const retentionInDays = !props.dataTraceLogRetention ? 30 : Math.max(props.dataTraceLogRetention.toDays(), 1);
+
+      // There is no parameter for CfnStage that will accept a non-default log group, however
+      // if the log group is created before API Gateway tries to create a log entry for a data trace
+      // log it'll use the log group that this creates.
+      const dataLoggingLogGroup = new CfnLogGroup(this, 'data-trace-logging-log-group', {
+        logGroupName: `/aws/apigateway/${props.webSocketApi.apiId}/${this.physicalName}`,
+        retentionInDays,
+      });
+
+      if (!props.dataTraceLogRemovalPolicy) {
+        dataLoggingLogGroup.applyRemovalPolicy(RemovalPolicy.DESTROY);
+      } else {
+        dataLoggingLogGroup.applyRemovalPolicy(props.dataTraceLogRemovalPolicy);
+      }
+    }
+
+    let destinationArn: string | undefined = undefined;
+    if (props.accessLogEnabled) {
+      if (!props.accessLogGroupArn) {
+        // We need to set up the right permissions to create the log group.
+        const iamRoleForLogGroup = new Role(this, 'IAMRoleForAccessLog', {
+          roleName: PhysicalName.GENERATE_IF_NEEDED,
+          assumedBy: new ServicePrincipal('apigateway.amazonaws.com'),
+        });
+        iamRoleForLogGroup.applyRemovalPolicy(RemovalPolicy.DESTROY);
+        iamRoleForLogGroup.node.addDependency(this.api);
+
+        iamRoleForLogGroup.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('service-role/AmazonAPIGatewayPushToCloudWatchLogs'));
+
+        // It's required to register the iam role that is used to create the log group with the account
+        // if an api gateway has never been created in this account/region before.  Otherwise, this is
+        // a no-op.
+        const account = new CfnAccount(this, 'account', {
```

Comment:
We should explicitly retain this since there can only be a single `CfnAccount` per environment.

Reply:
Done

Reply:
After thinking about this more, I think we should not create this for the user. I know we do in the `apigateway` module, but I don't think we should. Since this is a global property I think we should expect this to have been created already in a separate stack, and we should recommend this in the README.